pkg/reconciler/revision/resources/queue.go

```diff
@@ -82,6 +79,88 @@ var (
 	}
 )
 
+func getResources(annotations map[string]string, userContainer *corev1.Container) corev1.ResourceRequirements {
+
+	resources := corev1.ResourceRequirements{}
+	resourceRequests := corev1.ResourceList{corev1.ResourceCPU: queueContainerCPU}
+	resourceLimits := corev1.ResourceList{}
+	ok := false
+	var requestCPU, limitCPU, requestMemory, limitMemory resource.Quantity
+	var resourcePercentage float32
+
+	if ok, resourcePercentage = getResourcePercentageFromAnnotations(annotations, serving.QueueSideCarResourcePercentageAnnotation); ok {
+
+		if ok, requestCPU = getUserContainerResourceRequirements(userContainer.Resources.Requests.Cpu(), resourcePercentage, queueContainerRequestCPU); ok {
+			resourceRequests[corev1.ResourceCPU] = requestCPU
+		}
+
+		if ok, limitCPU = getUserContainerResourceRequirements(userContainer.Resources.Limits.Cpu(), resourcePercentage, queueContainerLimitCPU); ok {
+			resourceLimits[corev1.ResourceCPU] = limitCPU
+		}
+
+		if ok, requestMemory = getUserContainerResourceRequirements(userContainer.Resources.Requests.Memory(), resourcePercentage, queueContainerRequestMemory); ok {
+			resourceRequests[corev1.ResourceMemory] = requestMemory
+		}
+
+		if ok, limitMemory = getUserContainerResourceRequirements(userContainer.Resources.Limits.Memory(), resourcePercentage, queueContainerLimitMemory); ok {
+			resourceLimits[corev1.ResourceMemory] = limitMemory
+		}
+
+	}
+
+	resources.Requests = resourceRequests
```

Comment:
should this also have a `len()` check?

Reply:
resources.Requests will always have 1 element for back-compat as we set the cpu request currently to 25m.
`resourceRequests := corev1.ResourceList{corev1.ResourceCPU: queueContainerCPU}`

Reply:
I see, this is scoped to the queue-proxy.  That makes sense.